src/rulesets/Base/evalpoly.jl

```diff
@@ -62,13 +62,13 @@ if VERSION ≥ v"1.4"
     end
     function _evalpoly_intermediates(x, p)
         N = length(p)
-        @inbounds yn = one(x) * p[N]
+        @inbounds yn = one(x) * p[end]
         ys = similar(p, typeof(yn), N - 1)
         @inbounds ys[1] = yn
         @inbounds for i in 2:(N - 1)
-            ys[i] = muladd(x, ys[i - 1], p[N - i + 1])
+            ys[i] = muladd(x, ys[i - 1], p[end - i + 1])
         end
-        @inbounds y = muladd(x, ys[N - 1], p[1])
+        @inbounds y = muladd(x, ys[N - 1], p[begin])
```

Comment:
i don't think `p[begin]` was valid syntax in Julia v1.0 (which ChainRules still supports), but i think `firstindex` achieves the same:
```suggestion
        @inbounds y = muladd(x, ys[N - 1], p[firstindex(p)])
```

Reply:
Thanks, I just noticed that in testing.